### ISSUE TITLE
only generate short audio

### ISSUE THREAD

**Issue by jax-explorer** [kijai/ComfyUI-MMAudio#4]
input 5s video, only generated 2s audio

**Comment by kijai** [comment]
Input should be 24fps

**Comment by SoftologyPro** [comment]
Is there a way to adjust for any FPS video?
I tried a bunch of short videos I created with other Text-to-Video systems and they are not 24 fps.

**Comment by doogyhatts** [comment]
There is force rate setting in the video loading node. Can try it and see if works.
My audio output was somehow truncated to 7.68 seconds, despite the video input with 8 seconds at 24 fps.

**Comment by SUP3RMASS1VE** [comment]
can only generate 6 seconds. from my 10second clip

**Comment by GamingDaveUk** [comment]
> Input should be 24fps

Any way to adjust that? Many of us are using this with ai generated video's so 24fps is not the norm, its getting 6 seconds from the video but generating only 3 seconds.

**Comment by kijai** [comment]
> > Input should be 24fps
> 
> Any way to adjust that? Many of us are using this with ai generated video's so 24fps is not the norm, its getting 6 seconds from the video but generating only 3 seconds.

Actually I mistakently thought it's 24 but it seems it's actually 25 fps that mmaudio expects, it's hardcoded in and I have not tried to change it, assuming that's what the model is trained with.

**Comment by GamingDaveUk** [comment]
> > > Input should be 24fps
> > 
> > 
> > Any way to adjust that? Many of us are using this with ai generated video's so 24fps is not the norm, its getting 6 seconds from the video but generating only 3 seconds.
> 
> Actually I mistakently thought it's 24 but it seems it's actually 25 fps that mmaudio expects, it's hardcoded in and I have not tried to change it, assuming that's what the model is trained with.

Yeah the duration input is not used at all since my vids are 12fps I took the loaded duration and doupled it with a math node then feed it into duration... so this went from 6 seconds to 12, hoping that since the frame rate is half of 24 that would give me 6 seconds..... alas it still gave me 3.

Is what it is, can I suggest that in the example workflow you add a note that the duration is based on 24 frames per second and the duration node is not used. 

Still a cool thing to mess around with and thank you for creating the wrapper :) Hopefully the duration will be possible as it improves.

**Comment by kijai** [comment]
I really haven't had the time to look into this closer since the initial push, the duration input is kinda useless for us indeed since we can just feed it the video length we want.
